spacy/pipeline/span_ruler.py

```diff
@@ -0,0 +1,572 @@
+import warnings
+from typing import Optional, Union, List, Dict, Tuple, Iterable, Any, Callable
+from typing import Sequence, Set, cast
+from pathlib import Path
+import srsly
+
+from .pipe import Pipe
+from .spancat import spancat_score
+from ..training import Example
+from ..language import Language
+from ..errors import Errors, Warnings
+from ..util import ensure_path, SimpleFrozenList, registry
+from ..tokens import Doc, Span
+from ..matcher import Matcher, PhraseMatcher
+from .. import util
+
+PatternType = Dict[str, Union[str, List[Dict[str, Any]]]]
+
+
+@Language.factory(
+    "future_entity_ruler",
+    assigns=["doc.ents"],
+    default_config={
+        "phrase_matcher_attr": None,
+        "validate": False,
+        "overwrite_ents": False,
+        "scorer": {"@scorers": "spacy.entity_ruler_scorer.v1"},
+        "ent_id_sep": "__unused__",
+    },
+    default_score_weights={
+        "ents_f": 1.0,
+        "ents_p": 0.0,
+        "ents_r": 0.0,
+        "ents_per_type": None,
+    },
+)
+def make_entity_ruler(
+    nlp: Language,
+    name: str,
+    phrase_matcher_attr: Optional[Union[int, str]],
+    validate: bool,
+    overwrite_ents: bool,
+    scorer: Optional[Callable],
+    ent_id_sep: str,
+):
+    if overwrite_ents:
+        ents_filter = overwrite_overlapping_ents_filter
+    else:
+        ents_filter = preserve_existing_ents_filter
```

Comment:
I'm not sure whether the names will be intuitive/understandable to users. I wonder whether we should focus on the "priority" aspect instead --> something like `prioritize_matches_ents_filter` and `prioritize_existing_ents_filter`, though these names are equally long :/

Reply:
I get confused by `prioritize_matches_ents`.

- `prioritize_existing_ents_filter`
- `prioritize_new_ents_filter`

Reply:
I could live with those!